www/js/diary/LeafletView.jsx

```diff
@@ -3,21 +3,29 @@ import { angularize } from "../angular-react-helper";
 import { object, string } from "prop-types";
 import { View } from "react-native";
 
+const mapSet = new Set();
+export function invalidateMaps() {
+  mapSet.forEach(map => map.invalidateSize());
+}
+
```

Comment:
I referred back to the issue from before and looked at this reproduction case: https://github.com/e-mission/e-mission-docs/issues/111 - (instead of switching screens to "Diary", which no longer exists, I used "Dashboard")
Without this fix, the reproduction case was always the same as you saw. With the fix, the maps load fine every time. I haven't seen weird maps since then; so thus far, it seems like no hacks will be needed.

I basically just did what you wanted to do in the first place https://github.com/e-mission/e-mission-docs/issues/111#issuecomment-233184618, but because we were using `ui-leaflet` then, you didn't have access to the underlying maps. Now we do.